src/util/cacheManager.ts

```diff
@@ -0,0 +1,66 @@
+import { Message } from 'discord.js';
+import { journalCache, gratefulnessCache} from '../data/channelCaches'
+
+/**
+ * @param msg The message object.
+ * @param channel The channel whose cache needs to be updated. 'j' stands for Journal and 'g' for Gratefulness
+ */
+export function addEntry(msg : Message, channel : 'j'|'g') : void{
+    const cache = resolveCache(channel)
+    const userID = msg.author.id
+    const currTime = new Date() //NOAH: Is this supposed to a Date Object or something like moment()?
+    const datestr = currTime.toDateString()
+    cache.set(userID, datestr)
+}
+
+
+/**
+ * @param msg The message object.
+ * @param channel The channel whose cache needs to be queried. 'j' stands for Journal and 'g' for Gratefulness
+ */
+export function validiateEntry(msg : Message, channel: 'j'|'g') : boolean{
+    const cache = resolveCache(channel)
+    const userID = msg.author.id
+    const last_entry = cache.get(userID)
+    if(!last_entry){
+        return false
+    }
+
+    /* NOAH
+    This is where we need to compare the current and fetched Dates
+    */
+
+    const isValid : boolean = true  //Placeholder variable to represent the comparision output. Will be replaced by the actual Date comparision
+    return isValid
+}
+
+/**
+ * @param channel The channel whose cache needs to be cleaned. 'j' stands for Journal and 'g' for Gratefulness
+ */
+export function purgeCache(channel: 'j'|'g') : void{
```

Comment:
We'll also need to make sure that we have a place in which we run this function every 30 minutes or something.